cmd/notation/registry.go

```diff
@@ -6,48 +6,62 @@ import (
 	"net"
 	"net/http"
 
+	notationRegistry "github.com/notaryproject/notation-go/registry"
 	"github.com/notaryproject/notation/internal/trace"
 	"github.com/notaryproject/notation/internal/version"
 	loginauth "github.com/notaryproject/notation/pkg/auth"
+
 	"github.com/notaryproject/notation/pkg/configutil"
 	"oras.land/oras-go/v2/registry"
 	"oras.land/oras-go/v2/registry/remote"
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
-func getSignatureRepositoryClient(opts *SecureFlagOpts, reference string) (*remote.Repository, error) {
+type repository struct {
+	notationRegistry.Repository
+	remoteRepo *remote.Repository
+}
+
+func (r *repository) SetHttpDebugLog(debug bool) {
+	if !debug {
+		return
+	}
+	if authClient, ok := r.remoteRepo.Client.(*auth.Client); ok {
+		if authClient.Client == nil {
+			authClient.Client = http.DefaultClient
+		}
+		if authClient.Client.Transport == nil {
+			authClient.Client.Transport = http.DefaultTransport
+		}
+		authClient.Client.Transport = trace.NewTransport(authClient.Client.Transport)
+	}
+}
+
+func NewNotationRepository(remoteRepo *remote.Repository) *repository {
+	return &repository{
+		Repository: notationRegistry.NewRepository(remoteRepo),
+		remoteRepo: remoteRepo,
+	}
+}
+
+func getSignatureRepositoryClient(opts *SecureFlagOpts, reference string) (*repository, error) {
 	ref, err := registry.ParseReference(reference)
 	if err != nil {
 		return nil, err
 	}
 	return getRepositoryClient(opts, ref)
 }
 
-func getRepositoryClient(opts *SecureFlagOpts, ref registry.Reference) (*remote.Repository, error) {
+func getRepositoryClient(opts *SecureFlagOpts, ref registry.Reference) (*repository, error) {
```

Comment:
Why can't we pass `debug bool` to this function?

Reply:
If `debug bool` is not preferred by some critical reasons, we should be able to pass `ctx contxt.Context` to it for tracing.

Reply:
Added HttpDebugLog Option.